main.py

```diff
@@ -827,6 +833,142 @@ def _send_flow_mods(
                 self._send_napp_event(switch, flow, "pending")
         return flow_mods
 
+    def _send_flows_by_switch_from_request(
+        self,
+        request: Request,
+        command: str,
+    ) -> JSONResponse:
+        """Send flows in the respective switches from request."""
+        content_type_json_or_415(request)
+        content_dict = get_json_or_400(request, self.controller.loop)
+        if not isinstance(content_dict, dict):
+            raise HTTPException(400, detail=f"Invalid payload: {content_dict}")
+
+        if not any(content_dict):
+            result = "The request body doesn't have any flows"
+            raise HTTPException(400, detail=result)
+        
+        switches = []
+        for dpid in content_dict:
+            flows = content_dict[dpid]["flows"]
+            if not any(flows):
+                result = f"The switch {dpid} in the request body doesn't have any flows"
+                raise HTTPException(400, detail=result)
+            try:
+                validate_cookies_and_masks(flows, command)
+            except ValueError as exc:
+                raise HTTPException(400, detail=str(exc))
+            
+            switch = self.controller.get_switch_by_dpid(dpid)
+            if not switch:
+                return JSONResponse({"response": "dpid not found."}, status_code=404)
+            switches.append(switch)
+            
+        flows_by_switch_to_log(log.info, command, content_dict)
```

Comment:
Can we do better and avoid maintaining another utility slicing log func? This helper func isn't too small, not good having to maintain more than one here just due to minor payload differences.